core/modules/replica/tools/qserv-replica-master.cc

```diff
@@ -35,715 +40,322 @@
+            _replicationIntervalSec  (60),
+            _workerResponseTimeoutSec(60),
+            _workerEvictTimeoutSec   (3600),
+            _qservSyncTimeoutSec     (60),
+            _numReplicas             (0),
+            _numIter                 (0),
+            _purge          (false),
+            _forceQservSync (false),
+            _permanentDelete(false),
+            _failed         (false),
+            _log(LOG_GET(logger)) {
+
+        // Configure the command line parser
+
+        parser().option(
+            "health-probe-interval",
+            "interval (seconds) between iterations of the health monitoring probes",
+            _healthProbeIntervalSec
+        ).option(
+            "replication-interval",
+            "interval (seconds) between running the linear sequence of"
+            " actions: check - fix-up - replicate - re-balance",
+            _replicationIntervalSec
+        ).option(
+            "worker-response-timeout",
+            "maximum number of seconds to wait before giving up"
+            " on worker probes when checking for workers",
+            _workerResponseTimeoutSec
+        ).option(
+            "worker-evict-timeout",
+            "the maximum number of seconds to allow troubles workers to recover"
+            " from the last catastrophic event before evicting them from a cluster",
+            _workerEvictTimeoutSec
+        ).option(
+            "qserv-sync-timeout",
+            "the maximum number of seconds to wait before Qserv workers respond"
+            " to the synchronization requests before bailing out and proceeding"
+            " to the next step in the normal replication sequence. A value which"
+            " differs from 0 would override the corresponding parameter specified"
+            " in the Configuration.",
+            _qservSyncTimeoutSec
+        ).flag(
+            "qserv-sync-force",
+            "the flag which would force Qserv workers to update their list of replicas"
+            " even if some of the chunk replicas were still in use by on-going queries."
+            " This affect replicas to be deleted from the workers during the synchronization"
+            " stages",
+            _forceQservSync
+        ).option(
+            "replicas",
+            "the minimal number of replicas when running the replication phase"
+            " This number if provided and if it's not 0 will override the corresponding value found"
+            " in the Configuration.",
+            _numReplicas
+        ).option(
+            "iter",
+            "the number of iterations (a value of 0 means running indefinitely)",
+            _numIter
+        ).flag(
+            "purge",
+            "The binary flag which if provided would enable the optional purge algorithm in"
+            " the end of each replication cycle in order to eliminate excess replicas which"
+            " might get created by algorithm ran earlier in the cycle.",
```

Comment:
'an algorithm' or 'algorithms'.
If I may, this may be more clear:
The binary flag which, if provided, enables the 'purge' algorithm at the end of each replication cycle that eliminates excess replicas, which may have been created by algorithms executed earlier in the cycle.

Reply:
👍